src/librustc_middle/ty/mod.rs

```diff
@@ -195,22 +195,22 @@ pub struct AssocItem {
     pub container: AssocItemContainer,
 
     /// Whether this is a method with an explicit self
-    /// as its first argument, allowing method calls.
-    pub method_has_self_argument: bool,
+    /// as its first parameter, allowing method calls.
+    pub fn_has_self_parameter: bool,
 }
 
 #[derive(Copy, Clone, PartialEq, Debug, HashStable)]
 pub enum AssocKind {
     Const,
-    Method,
+    Fn,
     OpaqueTy,
     Type,
 }
 
 impl AssocKind {
     pub fn suggestion_descr(&self) -> &'static str {
         match self {
-            ty::AssocKind::Method => "method call",
+            ty::AssocKind::Fn => "method call",
             ty::AssocKind::Type | ty::AssocKind::OpaqueTy => "associated type",
             ty::AssocKind::Const => "associated constant",
```

Comment:
These don't seem consistent, where would "method call" and "associated constant" both work?.

Reply:
Do you mean we need to change it to "fn call" and "assoc"?

Reply:
I have no idea why it says "call", or why that makes sense.

Reply:
Maybe it use like: error: casts cannot be followed by a method call

Reply:
But that wouldn't make sense for the other variants, which is why I'm confused.

Okay I just checked and we should be able to change this to `"associated function"`. Which means `suggestion_descr` users could instead use `.as_def_kind().descr()`.

Reply:
Refactored.